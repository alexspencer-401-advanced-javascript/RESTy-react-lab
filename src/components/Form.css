.Form {
  font-family: Arial, Helvetica, sans-serif;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  background-color: lightgray;
  input[type="radio" ] {
    display: none;
  }
  label {
    margin-right: 6px;
    margin-left: 1em;
    cursor: pointer;
    background-color: navy;
    padding: 4px;
    color: white;
  }
  border: 1px dashed goldenrod;
}

.history {
  width: 30vw;
  margin: 1em;
  display: flex;
}


.jsonContainer {
  width: 70vw;
  margin: 1em;
  display: flex;
}

.results {
  background-color: white;
  width: 70vw;
  margin: 1em;
  padding: 5px;
}

.UrlInput {
  width: 70vw;
  margin: 1em;
}