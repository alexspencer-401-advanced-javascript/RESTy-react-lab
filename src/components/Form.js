import React, { Component } from 'react';
import apiCall from '../services/api-call';
import Results from './Results';

export default class Form extends Component {
  state = {
    apiData: {},
    url: '',
    textArea: ''
  }


  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    apiCall(this.state.url, this.state.name)
      .then(result => {
        console.log('Result: ', result);
        this.setState({ apiData: result });
      });
  }
  
  render() { //check textarea value
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <input type="text" name="url" value={this.state.url} onChange={this.handleChange} />
          <div>
            <input type="radio" name="method" value="GET" onChange={this.handleChange}/>GET
            <input type="radio" name="method" value="PUT" onChange={this.handleChange}/>PUT
            <input type="radio" name="method" value="POST" onChange={this.handleChange}/>POST
            <input type="radio" name="method" value="PATCH" onChange={this.handleChange}/>PATCH
            <input type="radio" name="method" value="DELETE" onChange={this.handleChange}/>DELETE
            <button>Go!</button>
          </div>
          <input type="textarea" value={this.state.textArea.value} onChange={this.handleChange} />
        </form>
        <Results results={this.state.result}/>
      </>
    );
  }
}
