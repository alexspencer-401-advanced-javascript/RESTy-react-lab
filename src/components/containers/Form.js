import React, { Component } from 'react';

export default class Form extends Component {
  state = {

  }

  handleChange = ({ target }) => {
    this.setState({value: target.value})
  }

  handleSubmit = (event) => {
    event.preventDefault();
  }

  render() { //check textarea value
    return (
      <form onSubmit={this.handleSubmit}>
        <input type="text" value={this.state.value} onChange={this.handleChange}/>
        <div>
        <input type="radio" name="GET" value="GET"/>GET
        <input type="radio" name="PUT" value="PUT"/>PUT
        <input type="radio" name="POST" value="POST"/>POST
        <input type="radio" name="PATCH" value="PATCH"/>PATCH
        <input type="radio" name="DELETE" value="DELETE"/>DELETE
        <button>Go!</button>
        </div>
        <input type="textarea" value={this.state.value} onChange={this.handleChange}/> 
      </form>
    );
  }
}