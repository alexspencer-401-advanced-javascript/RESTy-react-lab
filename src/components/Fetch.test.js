import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';


jest.mock('../services/api-call.js', () => ({
  apiCall() {
    return Promise.resolve([
      {
        '_id': '5da237699734fdcb7bef8f56',
        'name': 'Miles Shortman',
        'image': 'https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639'
      }]);
  }
}));

describe('api call fetch', () => {
  it('makes a successful api call and returns characters', () => {
    const wrapper = shallow(<Form />);
    expect(wrapper).toMatchSnapshot();
  });
});
