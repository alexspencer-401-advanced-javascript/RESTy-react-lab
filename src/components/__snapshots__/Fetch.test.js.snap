// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api call fetch makes a successful api call and returns characters 1`] = `
<div>
  <History
    history={Array []}
  />
  <form
    onSubmit={[Function]}
  >
    <input
      name="url"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div>
      <input
        name="method"
        onChange={[Function]}
        type="radio"
        value="GET"
      />
      GET
      <input
        name="method"
        onChange={[Function]}
        type="radio"
        value="PUT"
      />
      PUT
      <input
        name="method"
        onChange={[Function]}
        type="radio"
        value="POST"
      />
      POST
      <input
        name="method"
        onChange={[Function]}
        type="radio"
        value="PATCH"
      />
      PATCH
      <input
        name="method"
        onChange={[Function]}
        type="radio"
        value="DELETE"
      />
      DELETE
      <button>
        Go!
      </button>
    </div>
    <textarea
      cols="30"
      name="jsonInput"
      onChange={[Function]}
      rows="10"
      value=""
    />
  </form>
  <Results
    results={Array []}
  />
</div>
`;
